app/src/lib/stores/git-store.ts

```diff
@@ -372,6 +385,37 @@ export class GitStore extends BaseStore {
         .shift() || null
   }
 
+  public async fetchTagsToPush(account: IGitAccount | null) {
+    const currentRemote = this._currentRemote
+
+    if (currentRemote === null) {
+      this._tagsToPush = null
+      return
+    }
+
+    if (this.tip.kind !== TipState.Valid) {
+      this._tagsToPush = null
+      return
+    }
+    const currentBranch = this.tip.branch
+
+    const tagsToPush = await this.performFailableOperation(async () => {
+      const localTags = await this.getAllTags()
```

Comment:
this line will get called every time this method is called (via `_refreshRepository` for example). `_refreshRepository` is called from a _lot_ of different places, but i think we only need to load all the tags in the following cases:

* after a push/pull/fetch
* after the app is foregrounded
* when the user switches repos or branches

does that sound right? is it possible to only call this method in these cases?
_____
**more context:** `_refreshRepository` gets called from a lot of different places and its pretty expensive computationally. we've had performance issues in the past when we've added too many git calls (or other work) to it, so i'm being extra cautious here.

Reply:
Thanks for the tip!

Yup I think it's possible to just load the tags in any of these situations. I'm going to add this commit: https://github.com/desktop/desktop/pull/9638/commits/b2319c584e8392b7969a261270986b8fd8b36402 to this PR so we keep the list of local tags in memory, this way I'll be able to call `loadLocalTags` independently from `fetchTagsToPush`.